In [1]:
import pandas as pd
import numpy as np

In [2]:
df_inicial = pd.read_csv('../../data/crudos/bd_full_inicial_corte27jun.csv',
                         error_bad_lines=False, warn_bad_lines=False,
                         low_memory=False)
df_avanzado = pd.read_csv('../../data/crudos/bd_full_avanzado_corte27jun.csv',
                          error_bad_lines=False, warn_bad_lines=False,
                          low_memory=False)


df_inicial = df_inicial[df_inicial.Encuesta == 'U4 Encuesta de satisfacción (23 puntos)']
df_avanzado = df_avanzado[df_avanzado.Encuesta == 'UA4 Encuesta de satisfacción (3 puntos)']

In [3]:
# PIVOT DATA
pivot_inicial = df_inicial.pivot_table(
    index=['Nombre', 'Apellido', 'Correo Electrónico', 'Curso', 'ID Asignado Por Moodle', 'Nombre De Usuario'],
    columns='Pregunta', 
    values='Respuesta', 
    aggfunc='first').reset_index()

pivot_avanzado = df_avanzado.pivot_table(
    index=['Nombre', 'Apellido', 'Correo Electrónico', 'Curso', 'ID Asignado Por Moodle', 'Nombre De Usuario'],
    columns='Pregunta', 
    values='Respuesta', 
    aggfunc='first').reset_index()

In [4]:
# Remove \n in columns
pivot_inicial.columns = [col.replace("\n", " ").strip() 
                        for col in pivot_inicial.columns]
pivot_avanzado.columns = [col.replace("\n", " ").strip() 
                        for col in pivot_avanzado.columns]

## Limpieza de datos

In [5]:
drop_both = ['Nombre', 'Apellido', 'Correo Electrónico', 'ID Asignado Por Moodle', 'Número de cédula']
pivot_inicial.drop(columns=drop_both, inplace=True, errors='ignore')
pivot_avanzado.drop(columns=drop_both, inplace=True, errors='ignore')

In [6]:
# Eliminar fila cursed

# pivot_inicial.drop(445, axis=0, inplace=True) # Para v1

pivot_inicial.drop(489, axis=0, inplace=True) # Para v2 (datos actualizados hasta 27 jun)

In [7]:
pivot_inicial.to_excel("pivot_inicial_satis.xlsx", index=False, encoding='utf-8-sig')
pivot_avanzado.to_excel("pivot_avanzado_satis.xlsx", index=False, encoding='utf-8-sig')

### Veamos las columnas compartidas

In [8]:
# looking for common columns
#compartidas = set(pivot_inicial.columns).intersection(set(pivot_avanzado.columns))
#list(compartidas)

compartidas = {
    'Nombre De Usuario': 'Identificación',
    '¿Recibí los datos de acceso a la plataforma antes del inicio formal del curso?': '4. ¿Recibí los datos de acceso a la plataforma antes del inicio formal del curso?',
    '¿La información y las instrucciones para ingresar a la plataforma del curso fueron claras?': '5. ¿La información y las instrucciones para ingresar a la plataforma del curso fueron claras?',
    'Pude acceder de manera fácil e intuitiva a los contenidos y actividades del curso en la plataforma.': '6. Pude acceder de manera fácil e intuitiva a los contenidos y actividades del curso en la plataforma.',
    'Cuando lo solicité, obtuve apoyo de la Mesa de Ayuda para resolver mis dificultades con el acceso a la plataforma.': '7. Cuando lo solicité, obtuve apoyo de la Mesa de Ayuda para resolver mis dificultades con el acceso a la plataforma.',
    'Me siento satisfecho con el apoyo brindado por la Mesa de Ayuda.': '8. Me siento satisfecho con el apoyo brindado por la Mesa de Ayuda.',
    '¿Tiene alguna sugerencia para mejorar el servicio que presta la plataforma del curso y/o la Mesa de Ayuda? (opcional)': '9. ¿Tiene alguna sugerencia para mejorar el servicio que presta la plataforma del curso y/o la Mesa de Ayuda? (opcional)',
    # OJO: Pregunta 10 tiene diferentes opciones para inicial y avanzado
    'Por favor evalúe cada una de las fichas de la formación del 1 al 10, según su nivel de dificultad (10 es muy difícil)': '10. Por favor evalúe cada una de las fichas de la formación del 1 al 10, según su nivel de dificultad (10 es muy difícil)',
    # OJO: Pregunta 11 tiene diferentes opciones para inicial y avanzado
    'Por favor evalúe cada una de las fichas de la formación del 1 al 10, según su nivel de pertinencia (10 es muy pertinente)': '11. Por favor evalúe cada una de las fichas de la formación del 1 al 10, según su nivel de pertinencia (10 es muy pertinente)',
    'Agrega cualquier retroalimentación adicional que debamos considerar sobre las fichas (opcional)': '12. Agrega cualquier retroalimentación adicional que debamos considerar sobre las fichas (opcional)',
    'El lenguaje que se utiliza en los textos y videos de contenidos es claro y de fácil comprensión.': '13. El lenguaje que se utiliza en los textos y videos de contenidos es claro y de fácil comprensión.',
    'La cantidad de contenidos (videos y materiales de lectura) publicados en la plataforma cada semana, es adecuada.': '14. La cantidad de contenidos (videos y materiales de lectura) publicados en la plataforma cada semana, es adecuada.',
    'El número de tareas que se asignan para desarrollar cada semana es apropiado.': '15. El número de tareas que se asignan para desarrollar cada semana es apropiado.',
    'Considero que lo que he aprendido en el curso, lo puedo aplicar fácilmente en mi contexto de enseñanza/aprendizaje.': '16. Considero que lo que he aprendido en el curso, lo puedo aplicar fácilmente en mi contexto de enseñanza/aprendizaje.',
    'Los materiales y actividades del programa se adaptan a mi experiencia y conocimientos previos.': '18. Los materiales y actividades del programa se adaptan a mi experiencia y conocimientos previos.',
    '¿Con qué frecuencia, en una escala de 1-10 (donde 10 es siempre), los siguientes componentes del programa contribuyeron a que usted pudiera aprender y desarrollar habilidades?': '19. ¿Con qué frecuencia, en una escala de 1-10 (donde 10 es siempre), los siguientes componentes del programa contribuyeron a que usted pudiera aprender y desarrollar habilidades?',
    'En una escala de 1-10 (donde 1 es nada pertinente y 10 es muy pertinente), que nivel refleja mejor su percepción sobre la pertinencia de los contenidos abordados durante la formación en su labor como docente:': '20. En una escala de 1-10 (donde 1 es nada pertinente y 10 es muy pertinente), que nivel refleja mejor su percepción sobre la pertinencia de los contenidos abordados durante la formación en su labor como docente:',
    'De acuerdo con su respuesta anterior, explique su percepción sobre la pertinencia de los contenidos abordados durante la formación en su labor como docente': '21. De acuerdo con su respuesta anterior, explique su percepción sobre la pertinencia de los contenidos abordados durante la formación en su labor como docente',
    '¿En total cuánto tiempo dedicó semanalmente para desarrollar las actividades de formación (conectadas y desconectadas)? (respuesta restringida solamente a números)': '22. ¿En total cuánto tiempo dedicó semanalmente para desarrollar las actividades de formación (conectadas y desconectadas)? (respuesta restringida solamente a números)',
    '¿Tiene alguna sugerencia para mejorar los contenidos y/o actividades propuestas? (opcional)': '23. ¿Tiene alguna sugerencia para mejorar los contenidos y/o actividades propuestas? (opcional)',
    'Los encuentros sincrónicos comenzaron y finalizaron puntualmente.': '24. Los encuentros sincrónicos comenzaron y finalizaron puntualmente.',
    'Me sentí motivado(a) a asistir a los encuentros sincrónicos.': '25. Me sentí motivado(a) a asistir a los encuentros sincrónicos.',
    'Considero que las asesorías, los contenidos y actividades desarrolladas  durante los encuentros sincrónicos fueron apropiadas para el curso que  estoy haciendo.': '26. Considero que las asesorías, los contenidos y actividades desarrolladas  durante los encuentros sincrónicos fueron apropiadas para el curso que estoy haciendo.',
    '¿Tiene alguna sugerencia para mejorar los encuentros sincrónicos? (opcional)': '28. ¿Tiene alguna sugerencia para mejorar los encuentros sincrónicos? (opcional)',
    'Eliga una opción por pregunta, por favor.': '29. Elija una opción por pregunta, por favor.',
    '¿Tiene alguna sugerencia para mejorar el acompañamiento que el/la mentor(a) le brinda a sus docentes durante el curso? (opcional)': '30. ¿Tiene alguna sugerencia para mejorar el acompañamiento que el/la mentor(a) le brinda a sus docentes durante el curso? (opcional)',
    'El tipo de formación se adapta a mi forma de aprender.': '31. El tipo de formación se adapta a mi forma de aprender.',
    'Considero que los conocimientos y materiales adquiridos durante el programa son relevantes para mi trabajo como docente.': '32. Considero que los conocimientos y materiales adquiridos durante el programa son relevantes para mi trabajo como docente.',
    'Considero que los recursos de internet y equipos con los que cuento fueron suficientes para participar en las actividades del curso.': '33. Considero que los recursos de internet y equipos con los que cuento fueron suficientes para participar en las actividades del curso.',
    'Pude comprender los contenidos o materiales del curso sin mayores dificultades.': '34. Pude comprender los contenidos o materiales del curso sin mayores dificultades.',
    'Pude realizar las actividades del curso sin mayores dificultades.': '35. Pude realizar las actividades del curso sin mayores dificultades.',
    'Pude disponer, cabalmente, del tiempo semanal requerido para desarrollar las actividades propuestas de forma adecuada.': '36. Pude disponer, cabalmente, del tiempo semanal requerido para desarrollar las actividades propuestas de forma adecuada.',
    'Lea con atención las declaraciones y luego elija la opción que mejor refleje su punto de vista en cada caso.': '37. Lea con atención las declaraciones y luego elija la opción que mejor refleje su punto de vista en cada caso.',
    '¿Cuál cree que fue la mayor fortaleza del curso de formación? Describa aspectos relacionados con los contenidos, los encuentros sincrónicos, las tutorías, el acompañamiento de los mentores, los talleres, etc': '38. ¿Cuál cree que fue la mayor fortaleza del curso de formación? Describa aspectos relacionados con los contenidos, los encuentros sincrónicos, las tutorías, el acompañamiento de los mentores, los talleres, etc',
    '¿Cuál cree que fue la mayor debilidad del curso de formación? Describa aspectos relacionados con los contenidos, los encuentros sincrónicos, las tutorías, el acompañamiento de los mentores, los talleres, etc': '39. ¿Cuál cree que fue la mayor debilidad del curso de formación? Describa aspectos relacionados con los contenidos, los encuentros sincrónicos, las tutorías, el acompañamiento de los mentores, los talleres, etc',
    '¿Qué aspectos de la formación cree que deberían eliminarse o cambiarse? (describa aspectos relacionados con los contenidos, los encuentros sincrónicos, las tutorías, el acompañamiento de los mentores, los talleres, etc.)': '40. ¿Qué aspectos de la formación cree que deberían eliminarse o cambiarse? (describa aspectos relacionados con los contenidos, los encuentros sincrónicos, las tutorías, el acompañamiento de los mentores, los talleres, etc.)',
    'En una escala del 1 al 10 siendo 1 muy poco probable y 10 muy probable ¿Cuál es la probabilidad de que recomiende el British Council a un amigo o colega?': '41. En una escala del 1 al 10 siendo 1 muy poco probable y 10 muy probable ¿Cuál es la probabilidad de que recomiende el British Council a un amigo o colega?',
    'De acuerdo con su experiencia en el programa Coding For Kids en una escala del 1 al 5 siendo 1 el grado más bajo y 5 el más alto, ¿en qué medida ha cambiado su percepción favorable sobre el Reino Unido?': '42. De acuerdo con su experiencia en el programa Coding For Kids en una escala del 1 al 5 siendo 1 el grado más bajo y 5 el más alto, ¿en qué medida ha cambiado su percepción favorable sobre el Reino Unido?',
    '¿Qué estrategias y herramientas ha aprendido durante su formación en el proyecto Coding for kids para la implementación de una educación inclusiva y de calidad?': '43. ¿Qué estrategias y herramientas ha aprendido durante su formación en el proyecto Coding for kids para la implementación de una educación inclusiva y de calidad?',
    'Además del programa Coding for Kids ¿En qué actividades ha participado que le permitan fomentar e implementar una educación inclusiva y de calidad?': '44. Además del programa Coding for Kids ¿En qué actividades ha participado que le permitan fomentar e implementar una educación inclusiva y de calidad?',
    'Agradecemos sus comentarios con relación a si logramos cumplir o no con sus expectativas y en cuanto a la calidad del servicio recibido. (opcional)': '45. Agradecemos sus comentarios con relación a si logramos cumplir o no con sus expectativas y en cuanto a la calidad del servicio recibido. (opcional)',
}

In [9]:
inicial = {
    'Los contenidos y actividades del curso me brindaron herramientas básicas para empezar a desarrollar el pensamiento computacional': '17. Los contenidos y actividades del curso me brindaron herramientas básicas para empezar a desarrollar el pensamiento computacional',
    'los encuentros sincrónicos fueron desarrollados en forma clara y ordenada.': '27. Los encuentros sincrónicos fueron desarrollados en forma clara y ordenada.'
}

avanzado = {
    'Los contenidos y actividades del curso me brindaron herramientas más avanzadas para desarrollar el pensamiento computacional de mis estudiantes.': '17. Los contenidos y actividades del curso me brindaron herramientas más avanzadas para desarrollar el pensamiento computacional de mis estudiantes.',
    'Los encuentros sincrónicos fueron desarrollados en forma clara y ordenada.': '27. Los encuentros sincrónicos fueron desarrollados en forma clara y ordenada.'
}

In [10]:
# Arregar estas columnas 
col = "¿En total cuánto tiempo dedicó semanalmente para desarrollar las actividades de formación (conectadas y desconectadas)? (respuesta restringida solamente a números)"
col_aux = "¿En total cuánto tiempo (en horas) dedicó semanalmente para desarrollar las actividades de formación (conectadas y desconectadas)? (respuesta restringida solamente a números)"


In [11]:
pivot_inicial[col].to_frame().info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4179 entries, 0 to 4179
Data columns (total 1 columns):
 #   Column                                                                                                                                                               Non-Null Count  Dtype 
---  ------                                                                                                                                                               --------------  ----- 
 0   ¿En total cuánto tiempo dedicó semanalmente para desarrollar las actividades de formación (conectadas y desconectadas)? (respuesta restringida solamente a números)  4131 non-null   object
dtypes: object(1)
memory usage: 65.3+ KB


In [12]:
df_col = pivot_inicial.loc[:, col].to_frame()
df_aux = pivot_inicial.loc[:, col_aux].to_frame().rename(columns={col_aux: col})
pivot_inicial[col] = df_col.combine_first(df_aux)[col]

pivot_inicial.drop(columns=col_aux, inplace=True)

In [13]:
pivot_inicial[col].to_frame().info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4179 entries, 0 to 4179
Data columns (total 1 columns):
 #   Column                                                                                                                                                               Non-Null Count  Dtype 
---  ------                                                                                                                                                               --------------  ----- 
 0   ¿En total cuánto tiempo dedicó semanalmente para desarrollar las actividades de formación (conectadas y desconectadas)? (respuesta restringida solamente a números)  4177 non-null   object
dtypes: object(1)
memory usage: 65.3+ KB


Se pasan de 3705 a 3746 valores no nulos. 😎


Para los datos actualizados hasta el 27 de jun:

    Se pasan de 4131 a 4177 valores no nulos. 😎

In [14]:
pivot_inicial.rename(columns=compartidas, inplace=True)
pivot_inicial.rename(columns=inicial, inplace=True)
pivot_inicial.to_excel('pivot_inicial_renamed.xlsx', index=False, encoding='utf-8-sig')

pivot_avanzado.rename(columns=compartidas, inplace=True)
pivot_avanzado.rename(columns=inicial, inplace=True)
pivot_avanzado.to_excel('pivot_avanzado_renamed.xlsx', index=False, encoding='utf-8-sig')


## Limpieza por pregunta

In [15]:
def add_columns(df1, df2):
    for col in df2.columns:
        df1[col] = df2[col]

de_acuerdo= {
    1: 'Totalmente en desacuerdo', 
    2: 'En desacuerdo',
    3: 'Neutro',
    4: 'De acuerdo',
    5: 'Totalmente de acuerdo'
}

In [16]:
## Pregunta 10 inicial
col = "10. Por favor evalúe cada una de las fichas de la formación del 1 al 10, según su nivel de dificultad (10 es muy difícil)"
df_res = pivot_inicial[col].str.split(r'\b\D+\b', expand=True)
df_res
df_res.rename({
    2: '10.1 Ficha 0: Orientaciones',
    4: '10.2 Ficha 1: Luces y Códigos',
    6: '10.3 Ficha 2: Salvando las Tortugas',
    8: '10.4 Ficha 3: Expedición espeleológica',
    10: '10.5 Ficha 4: Incertidumbre cierta'
}, axis=1, inplace=True)
df_res.drop([0, 1, 3, 5, 7, 9], axis=1, inplace=True)
print(pivot_inicial.shape)
add_columns(pivot_inicial, df_res)
pivot_inicial.drop(columns=col, inplace=True)
print(pivot_inicial.shape)

(4179, 46)
(4179, 50)


In [17]:
## Pregunta 10 avanzado
col = "10. Por favor evalúe cada una de las fichas de la formación del 1 al 10, según su nivel de dificultad (10 es muy difícil)"
df_res = pivot_avanzado[col].str.split(r'\b\D+\b', expand=True)
df_res
df_res.rename({
    2: '10.1 Ficha 0: Orientaciones',
    4: '10.2 Ficha 7: Misión a Marte',
    6: '10.3 Ficha 8: Invernaderos',
    8: '10.4 Ficha 9: Computación y virus',
    10: '10.5 Ficha 10: Un algoritmo para ir a la escuela'
}, axis=1, inplace=True)
df_res.drop([0, 1, 3, 5, 7, 9], axis=1, inplace=True)
print(pivot_avanzado.shape)
add_columns(pivot_avanzado, df_res)
pivot_avanzado.drop(columns=col, inplace=True)
print(pivot_avanzado.shape)

(772, 46)
(772, 50)


In [18]:
## Pregunta 11 inicial
col = "11. Por favor evalúe cada una de las fichas de la formación del 1 al 10, según su nivel de pertinencia (10 es muy pertinente)"
df_res = pivot_inicial[col].str.split(r'\b\D+\b', expand=True)
df_res
df_res.rename({
    2: '11.1 Ficha 0: Orientaciones',
    4: '11.2 Ficha 1: Luces y Códigos',
    6: '11.3 Ficha 2: Salvando las Tortugas',
    8: '11.4 Ficha 3: Expedición espeleológica',
    10: '11.5 Ficha 4: Incertidumbre cierta'
}, axis=1, inplace=True)
df_res.drop([0, 1, 3, 5, 7, 9], axis=1, inplace=True)
print(pivot_inicial.shape)
add_columns(pivot_inicial, df_res)
pivot_inicial.drop(columns=col, inplace=True)
print(pivot_inicial.shape)

(4179, 50)
(4179, 54)


In [19]:
## Pregunta 11 avanzado
col = "11. Por favor evalúe cada una de las fichas de la formación del 1 al 10, según su nivel de pertinencia (10 es muy pertinente)"
df_res = pivot_avanzado[col].str.split(r'\b\D+\b', expand=True)
df_res
df_res.rename({
    2: '11.1 Ficha 0: Orientaciones',
    4: '11.2 Ficha 7: Misión a Marte',
    6: '11.3 Ficha 8: Invernaderos',
    8: '11.4 Ficha 9: Computación y virus',
    10: '11.5 Ficha 10: Un algoritmo para ir a la escuela'
}, axis=1, inplace=True)
df_res.drop([0, 1, 3, 5, 7, 9], axis=1, inplace=True)
print(pivot_avanzado.shape)
add_columns(pivot_avanzado, df_res)
pivot_avanzado.drop(columns=col, inplace=True)
print(pivot_avanzado.shape)

(772, 50)
(772, 54)


In [20]:
## Pregunta 19 inicial
col = "19. ¿Con qué frecuencia, en una escala de 1-10 (donde 10 es siempre), los siguientes componentes del programa contribuyeron a que usted pudiera aprender y desarrollar habilidades?"
df_res = pivot_inicial[col].str.split(r'\b\D+\b', expand=True)
df_res
df_res.rename({
    1: '19.1 Videos',
    2: '19.2 Fichas Didácticas',
    3: '19.3 Mentores',
    4: '19.4 Foros de Contenido',
    5: '19.5 Foros Taller',
    6: '19.6 Evidencias de Implementación docentes (wiki)',
    7: '19.7 Evidencias de Implementación Estudiantes (foro)',
    8: '19.8 Foros de interacción con otros participantes',
    9: '19.9 Los demás participantes',
}, axis=1, inplace=True)
df_res.drop([0], axis=1, inplace=True)
print(pivot_inicial.shape)
add_columns(pivot_inicial, df_res)
pivot_inicial.drop(columns=col, inplace=True)
print(pivot_inicial.shape)

(4179, 54)
(4179, 62)


In [21]:
## Pregunta 19 avanzado
col = "19. ¿Con qué frecuencia, en una escala de 1-10 (donde 10 es siempre), los siguientes componentes del programa contribuyeron a que usted pudiera aprender y desarrollar habilidades?"
df_res = pivot_avanzado[col].str.split(r'\b\D+\b', expand=True)
df_res
df_res.rename({
    1: '19.1 Videos',
    2: '19.2 Fichas Didácticas',
    3: '19.3 Mentores',
    4: '19.4 Foros de Contenido',
    5: '19.5 Foros Taller',
    6: '19.6 Evidencias de Implementación docentes (wiki)',
    7: '19.7 Evidencias de Implementación Estudiantes (foro)',
    8: '19.8 Foros de interacción con otros participantes',
    9: '19.9 Los demás participantes',
}, axis=1, inplace=True)
df_res.drop([0], axis=1, inplace=True)
print(pivot_avanzado.shape)
add_columns(pivot_avanzado, df_res)
pivot_avanzado.drop(columns=col, inplace=True)
print(pivot_avanzado.shape)

(772, 54)
(772, 62)


In [22]:
## Pregunta 29 inicial
col = "29. Elija una opción por pregunta, por favor."
df_res = pivot_inicial[col].str.split(r'\b\D+\b', expand=True)
df_res.iloc[:, 1:] = df_res.iloc[:, 1:].astype(int).replace(de_acuerdo)

df_res
df_res.rename({
    1: '29.1 La retroalimentación recibida por parte del/de la mentor(a) fue relevante',
    2: '29.2 El/La mentor(a) resolvía las dudas de los participantes de manera adecuada en los foros de discusión y en las sesiones sincrónicas',
    3: '29.3 El/La mentor(a) tuvo en cuenta los diferentes niveles de experiencia y habilidad de los participantes para el desarrollo de las sesiones sincrónicas',
    4: '29.4 El/La mentor(a) demostraba suficiencia y preparación para el desarrollo de las sesiones',
    5: '29.5 El acompañamiento realizado por el/la mentor(a) fue adecuado, pues considero que estuvo atento(a) a resolver mis dudas y dificultades',
    6: '29.6 Me siento satisfecho con el tiempo empleado por el/la mentor(a) en los espacios de interacción y retroalimentación.',
    7: '29.7 La retroalimentación que el/la mentor(a) me ha dado es útil y acorde a los requerimientos del curso.',
    8: '29.8 Gracias al acompañamiento de mi mentor(a) he podido superar dificultades con los temas y actividades propuestos en las fichas, que por mi mismo(a) no habría superado.',
    9: '29.9 Gracias al acompañamiento de mi mentor(a) me siento inspirado(a) para hacer cambios en mi práctica pedagógica.',
}, axis=1, inplace=True)
df_res.drop([0], axis=1, inplace=True)


print(pivot_inicial.shape)
add_columns(pivot_inicial, df_res)
pivot_inicial.drop(columns=col, inplace=True)
print(pivot_inicial.shape)



(4179, 62)
(4179, 70)


In [23]:
## Pregunta 29 avanzado
col = "29. Elija una opción por pregunta, por favor."
df_res = pivot_avanzado[col].str.split(r'\b\D+\b', expand=True)
df_res.iloc[:, 1:] = df_res.iloc[:, 1:].astype(int).replace(de_acuerdo)
df_res
df_res.rename({
    1: '29.1 La retroalimentación recibida por parte del/de la mentor(a) fue relevante',
    2: '29.2 El/La mentor(a) resolvía las dudas de los participantes de manera adecuada en los foros de discusión y en las sesiones sincrónicas',
    3: '29.3 El/La mentor(a) tuvo en cuenta los diferentes niveles de experiencia y habilidad de los participantes para el desarrollo de las sesiones sincrónicas',
    4: '29.4 El/La mentor(a) demostraba suficiencia y preparación para el desarrollo de las sesiones',
    5: '29.5 El acompañamiento realizado por el/la mentor(a) fue adecuado, pues considero que estuvo atento(a) a resolver mis dudas y dificultades',
    6: '29.6 Me siento satisfecho con el tiempo empleado por el/la mentor(a) en los espacios de interacción y retroalimentación.',
    7: '29.7 La retroalimentación que el/la mentor(a) me ha dado es útil y acorde a los requerimientos del curso.',
    8: '29.8 Gracias al acompañamiento de mi mentor(a) he podido superar dificultades con los temas y actividades propuestos en las fichas, que por mi mismo(a) no habría superado.',
    9: '29.9 Gracias al acompañamiento de mi mentor(a) me siento inspirado(a) para hacer cambios en mi práctica pedagógica.',
}, axis=1, inplace=True)
df_res.drop([0], axis=1, inplace=True)


print(pivot_avanzado.shape)
add_columns(pivot_avanzado, df_res)
pivot_avanzado.drop(columns=col, inplace=True)
print(pivot_avanzado.shape)



(772, 62)
(772, 70)


In [24]:
## Pregunta 37 inicial
col = "37. Lea con atención las declaraciones y luego elija la opción que mejor refleje su punto de vista en cada caso."
df_res = pivot_inicial[col].str.split(r'\b\D+\b', expand=True)
df_res.iloc[:, 1:] = df_res.iloc[:, 1:].astype(int).replace(de_acuerdo)
df_res
df_res.rename({
    1:   '37.1 Este curso cumplió con mis expectativas.',
    2:   '37.2 Este curso fue de buena calidad.',
    3:   '37.3 Adquirí nuevos conocimientos y/o habilidades por haber participado en este curso.',
    4:   '37.4 Los objetivos del programa se lograron.',
    5:   '37.5 Los materiales y actividades del curso fueron suficientes para alcanzar los objetivos propuestos.',
    6:   '37.6 Haber participado en el programa me ha motivado a buscar más información por mi propia cuenta.',
    7:   '37.7 El tiempo es adecuado para desarrollar las sesiones y las actividades propuestas.',
    8:   '37.8 El programa de formación tiene en cuenta las necesidades de capacitación y formación de los y las docentes que participamos.',
    9:   '37.9 Recomendaría este programa de formación a otros colegas docentes.',
    10: '37.10 Es gratificante participar en los proyectos impulsados por el British Council.',
    11: '37.11 Los esfuerzos realizados por el British Council a través de este tipo de proyectos son pertinentes para la educación en Colombia.',
    12: '37.12 Considero que el British Council se esfuerza por el fortalecimiento de la educación y la pedagogía en el país.',
    13: '37.13 Considero que Reino Unido está comprometido por el mejoramiento de la educación de las personas.',
}, axis=1, inplace=True)
df_res.drop([0], axis=1, inplace=True)
print(pivot_inicial.shape)
add_columns(pivot_inicial, df_res)
pivot_inicial.drop(columns=col, inplace=True)
print(pivot_inicial.shape)

(4179, 70)
(4179, 82)


In [25]:
## Pregunta 37 avanzado
col = "37. Lea con atención las declaraciones y luego elija la opción que mejor refleje su punto de vista en cada caso."
df_res = pivot_avanzado[col].str.split(r'\b\D+\b', expand=True)
df_res.iloc[:, 1:] = df_res.iloc[:, 1:].astype(int).replace(de_acuerdo)
df_res
df_res.rename({
    1:   '37.1 Este curso cumplió con mis expectativas.',
    2:   '37.2 Este curso fue de buena calidad.',
    3:   '37.3 Adquirí nuevos conocimientos y/o habilidades por haber participado en este curso.',
    4:   '37.4 Los objetivos del programa se lograron.',
    5:   '37.5 Los materiales y actividades del curso fueron suficientes para alcanzar los objetivos propuestos.',
    6:   '37.6 Haber participado en el programa me ha motivado a buscar más información por mi propia cuenta.',
    7:   '37.7 El tiempo es adecuado para desarrollar las sesiones y las actividades propuestas.',
    8:   '37.8 El programa de formación tiene en cuenta las necesidades de capacitación y formación de los y las docentes que participamos.',
    9:   '37.9 Recomendaría este programa de formación a otros colegas docentes.',
    10: '37.10 Es gratificante participar en los proyectos impulsados por el British Council.',
    11: '37.11 Los esfuerzos realizados por el British Council a través de este tipo de proyectos son pertinentes para la educación en Colombia.',
    12: '37.12 Considero que el British Council se esfuerza por el fortalecimiento de la educación y la pedagogía en el país.',
    13: '37.13 Considero que Reino Unido está comprometido por el mejoramiento de la educación de las personas.',
}, axis=1, inplace=True)
df_res.drop([0], axis=1, inplace=True)
print(pivot_avanzado.shape)
add_columns(pivot_avanzado, df_res)
pivot_avanzado.drop(columns=col, inplace=True)
print(pivot_avanzado.shape)

(772, 70)
(772, 82)


In [26]:
pivot_inicial.to_excel('enc_satisfaccion_inicial_v2.xlsx', index=False, encoding='utf-8-sig')
pivot_avanzado.to_excel('enc_satisfaccion_avanzado_v2.xlsx', index=False, encoding='utf-8-sig')